SteamBot/Logging/Log.cs

```diff
@@ -95,10 +99,22 @@ public void RemoveLoggingObject(LoggerBase logger)
 
         public void Dispose()
         {
-            Disposed = true;
-            foreach (IDisposable logger in LoggerObjects.OfType<IDisposable>())
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!Disposed)
             {
-                logger.Dispose();
+                if (disposing)
+                {
+                    foreach (IDisposable disposableLogger in LoggerObjects.OfType<IDisposable>())
```

Comment:
Is it alright if I assume Logging instances are managed resources? Probably should add note it's up to logger coder to make sure that his code frees up unmanaged resources when GC gets to it or when it's disposed.